### DISCUSSION TITLE
General

### DISCUSSION THREAD

**Discussion by OcelotWalrus** [OcelotWalrus/Cromha-Expansion-plugin#122] [General]
You can talk about various stuff about the plugin here and ask questions!

**Comment by OcelotWalrus** [comment]
I added a cool thing:

You may noticed but, there is a small region here, governed by the Cromha Empire:
![image](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/44a0254f-947d-41cf-a398-a52e63238c94)
It's where the Cromha Empire do some advanced and mysterious research but I'm still working on the lore of this. Anyway, the only way to access it is by passing with a Jump Drive through the Saykain system. Actually on the Saykain system, there is 3 Black Manta that are named after the 3 Drak Archons. These Black Manta will always stay in the system and destroy any ship that will enter this system that are not friendly.

**Reply by OcelotWalrus** [reply]
Just so you know.

**Reply by lumbar527** [reply]
So the Lumbarian Empire probably can't invade it, and it'll be a top-secret highly-defended place for the Cromha Empire to retreat to?

**Reply by OcelotWalrus** [reply]
Yes, but the won't ever retreat here: it's a whole military/research basement for the Cromha Empire.

It's where they designed and imagined their powerful Draining Field Devices, Anti-Missile Field Devices and other extremely powerful antimatter weapons like the Prototype Fusion Cannon, the Dragonflame Cannon and the Caltalyctic Tear and all other new antimatter weapons that I will add in the future.

**Reply by OcelotWalrus** [reply]
The only place that is governed by The Empire that the Lumbarian Empire could invade is the small systems in the Syndicate Space. But I think in the future I'll add system that are directly connected to the human space that will be governed by the Cromha Empire and Quargs.

Quargs and Cromha Empire are kind of the police of the space. They both describe themselves as 'Drak servitors', even if the Cromha Empire is at a higher tier of the Drak.

**Reply by lumbar527** [reply]
But it's unassailable, basically.

**Reply by lumbar527** [reply]
Why couldn't the Lumbarian Empire invade Empire Space? They have Sun Drives.

**Reply by OcelotWalrus** [reply]
Sure, but they would have to fight the Hai first, and if the Lumbarian Empire enter Hai space, the whole Great Alliance will call a special order to call all their fleets (Wanderers, Empire, Hai, Quarg, Remnant (but not Far Remnant) and I may forgot more)

**Reply by lumbar527** [reply]
The G[]s[]'[]a[] are powerful and plentiful, but one question:

Why are the Wanderers in the Great Alliance? They help (serve?) the Pug, and the Pug and Quarg hate each other.

**Reply by OcelotWalrus** [reply]
Well first, to share technologies and to help them with the Unfettered Hais. The main reason why The Empire did this is because there were close together.

**Reply by OcelotWalrus** [reply]
And even if the Quarg hates the Pug, they are not hostile to the Wanderer in any way.

**Comment by OcelotWalrus** [comment]
> I added a cool thing:
>
>You may noticed but, there is a small region here, governed by the Cromha Empire:
![image](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/44a0254f-947d-41cf-a398-a52e63238c94)
It's where the Cromha Empire do some advanced and mysterious research but I'm still working on the lore of this. Anyway, the only way to access it is by passing with a Jump Drive through the Saykain system. Actually on the Saykain system, there is 3 Black Manta that are named after the 3 Drak Archons. These Black Manta will always stay in the system and destroy any ship that will enter this system that are not friendly.

By the way, they are all Uninhabited system unless Kor Flas. But there is still cromha fleets and mineria transports fleets in the uninhabited systems.

**Comment by OcelotWalrus** [comment]
> Why couldn't the Lumbarian Empire invade Empire Space? They have Sun Drives.

I think what would be the best thing to do for the Lumbarian Empire is to secretly enter Wanderer Space without being caught by entering Uninhabited systems only and then entering the Far Empire Space. But still then, they will have to fight huge Cromha Empire fleets and even, at the Ring Wartrue system, there is the Great Alliance ringworld and they would have to fight Great Alliance fleets. And even if they happen to defeat these fleets, Lifdot will send deadly Cromha Protectionist Fleets very fast to the Lumbarian invaders.

**Comment by OcelotWalrus** [comment]
I think instead of they invade each others, they would happen to make a treaty, the Cromha Empire will help the Lumbarian Empire to develop, discover a new space region where the Lumbarian Empire will settle.

But how would they make the treaty? I don't have ideas for now.

**Reply by OcelotWalrus** [reply]
I think I may know:

The Far Empire Space is not finished at all. When I will expand this region in the future, I will add like a dozens of Empire Rebels systems.

So what will happen in the storyline is that the Cromha Empire is promise to the Lumbarian Empire to give them all the Empire Rebels systems and planets in the Far Empire Space. But the Lumbarian Empire will have to give tributes to the Cromha Empire every year. It will be 35% of what the systems the Cromha Empire give to the Lumbarian made money.

**Reply by OcelotWalrus** [reply]
And maybe the Cromha Empire will also give some knowledge to the Lumbarian Empire about the whole story of the Milky Way, but that's extra.

**Reply by lumbar527** [reply]
The Lumbarian Empire will never pay tribute! They shall have victory! Also, the Lumbarian Empire doesn't need more territory; they have the whole of human space!

**Comment by OcelotWalrus** [comment]
We should agree on one thing:

So there is the Empire storyline and the Lumbarian Empire storyline. So both of these storylines are just missions that follows each others that can help the empire by adding new weapons, or new ships but it can't affect the story in any way(the Main storyline).

And there is the Main storyline, it's where there will be events that lets you know about the war between The Empire and the Lumbarian Empire, you may be able to join fleets while the both empire fights but you won't really help any of the Empire compared to The Empire and the Lumbarian Empire storyline. 

And finally, the player won't be able to do both of the Cromha and the Lumbarian storyline.

**Reply by lumbar527** [reply]
Yep, that makes sense, if I understand what you're saying.

So, there is the Main storyline, which is not really a storyline but a series of events and a few missions that don't really matter.

And then with the Empire/Lumbarian Empire storyline, you can actually make a difference in the outcome, but maybe not common events like a few background battles?

**Reply by OcelotWalrus** [reply]
The Main storyline matters because the end of it means the end of the war.

**Reply by lumbar527** [reply]
Are you saying that the ending will always be the same? I'm not following, sorry.

**Comment by lumbar527** [comment]
Do you have any good metal textures for applying on ships/outfits? (Note, it may not work as well for me because I don't use GIMP)

**Reply by OcelotWalrus** [reply]
It can work with all software. What you do is basically add metal texture on top of your ship sprite layer then you lower the opacity of the layer where there is the metal texture. You also need to crop the layer texture so it is the same size as the outfit/ship.

**Reply by OcelotWalrus** [reply]
> Do you have any good metal textures for applying on ships/outfits? 

You can found metal textures in the source/ directory of the plugin. The metal textures are in .xcf files (GIMP files).

> (Note, it may not work as well for me because I don't use GIMP)

If you're not sure about what you're doing, maybe give me the render and I'll try to add metal textures to it.

**Reply by lumbar527** [reply]
I use Pixelmator Pro to do stuff and I know how to add the textures to stuff using that but I can't find good ones. So I don't have GIMP but all I need is an image (so far used .jpg and .png as textures, so they work).

**Reply by lumbar527** [reply]
I found something that works, I think.

**Comment by OcelotWalrus** [comment]
I thought about something:
![Screenshot from 2023-07-27 18-19-01](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/4762884f-97c4-429c-a518-1f646ade41b2)
What if the Lumbarian Empire will try to colonize these systems in the Gegno Space?

**Reply by lumbar527** [reply]
Sure, and the it turns out the Lumbarian Empire mostly wants to colonize areas and then they make peace with the Cromha Empire/Great Alliance because they only made combat stuff because they were worried about being attacked, but then they grew and became combat focused, but then the Main storyline will make them remember that they're not a conquering force, they're a colonizing force and then they would join the Great Alliance?

**Reply by OcelotWalrus** [reply]
Great idea!

**Reply by lumbar527** [reply]
Thanks! Do you think the Quarg would want them to, like, stop using their tech or would they make an agreement?

**Reply by OcelotWalrus** [reply]
I think what the Quarg will do is that since the Lumbarian Empire technically know how to reproduce them but they don't have enough advanced technologies to reproduce them so the Quarg will agree to sell to them only some of their less advanced outfits (but still very advanced for Lumbarian Empire).

**Reply by lumbar527** [reply]
I'm confused, since the Lumbarian Empire managed to reproduce them. They're good at figuring out how stuff works to revers-engineer them. They use Wardragons in their fleets.

**Reply by OcelotWalrus** [reply]
They just stolen them.

**Comment by OcelotWalrus** [comment]
I got an other idea!

The colonies in the Gegno Space are forgotten Lumbarian colonies that the Lumbarian Empire fought were dead but, when exploring this region of the space, some Quarg (since they already have a ringworld in the Gegno Space) will find the colonies. Some stuff will happen and the colonies will finally happen to make remember the Lumbarian Emperor about their original ideals and personality and would make the Lumbarian Empire stop to fight against the Cromha Empire.

**Reply by OcelotWalrus** [reply]
So I think the Lumbarian planets in the Gegno Space will be already in and there will be no events to make them "spawn".

**Reply by lumbar527** [reply]
Cool!

**Comment by OcelotWalrus** [comment]
> Yep, that makes sense, if I understand what you're saying.
>
>So, there is the Main storyline, which is not really a storyline but a series of events and a few missions that don't really matter.
>
>And then with the Empire/Lumbarian Empire storyline, you can actually make a difference in the outcome, but maybe not common events like a few background battles?

I think you did not understood.

The role of the Cromha and Lumbarian storyline is to join one of the faction, get Licenses and help them create outfits and ships and help them in some battles but nothing more. So even if you helped one of the empire to get powerful weapons, it won't affect in any way the Main storyline, aka the story of the war between the two empires. The Cromha and the Lumbarian storyline shall not affect the map with events by too big changes. (Changing a system of government or adding a station to an uninhabited system is accepted but invading a whole space region is not). 

While the role of the Main storyline is to make the lore as well as the story develop, to finally make the two Empires friendly. (Basically) Of course, sometimes, the player will be able to join one of the Empire side and to fight against other fleets.

**Reply by OcelotWalrus** [reply]
You'll need to re-write the Lumbarian Storyline and make it kind of like the Cromha Storyline. You can check the empire storylines plugin files to have examples of missions.

**Reply by lumbar527** [reply]
So, the Main storyline is a bunch of things that must happen, and the Empire/Lumbarian Empire storylines is you doing stuff that won't change the outcome?

**Reply by OcelotWalrus** [reply]
Yes, the goal of the Cromha and the Lumbarian storyline is to get licenses, help developing new technologies, fighting against hostile species in battles, escorting huge transports fleets, so missions.

**Reply by OcelotWalrus** [reply]
You can say that the Main storyline is a bunch of events and the other a bunch of missions. The other storylines are parallels to the Main one.

**Comment by OcelotWalrus** [comment]
> I got an other idea!
>
>The colonies in the Gegno Space are forgotten Lumbarian colonies that the Lumbarian Empire fought were dead but, when exploring this region of the space, some Quarg (since they already have a ringworld in the Gegno Space) will find the colonies. Some stuff will happen and the colonies will finally happen to make remember the Lumbarian Emperor about their original ideals and personality and would make the Lumbarian Empire stop to fight against the Cromha Empire.

I will take care of modifying these systems.

**Comment by OcelotWalrus** [comment]
So, what about the ship you said you was designing recently?

**Reply by lumbar527** [reply]
I'm possibly almost done.

**Reply by OcelotWalrus** [reply]
Nice!

**Reply by lumbar527** [reply]
Here's an image (not scaled properly yet, just scaled for GitHub loading):
![gus'pas chisel scaled](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/125756761/390e7e44-ec7a-4186-94a3-05c6ae081db1)

**Reply by OcelotWalrus** [reply]
Looking good.

**Reply by lumbar527** [reply]
Thanks! The <aliens> like to use heavy missile artillery, so that's why the gun ports are so largo.

**Reply by lumbar527** [reply]
Thumb: 
![gus'pas chisel thumb](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/125756761/dbbc5c45-fc58-47db-9ed4-f27b2b2063b5)

**Reply by lumbar527** [reply]
Now I have to figure out another ship to make, or maybe I could make some outfits.

By the way, I'm sort of hoping I can add an alien to vanilla. Once this develops more, is it OK with you if I try to PR to Endless Sky?

**Reply by OcelotWalrus** [reply]
Sure why are you asking me?

**Reply by lumbar527** [reply]
Just because this will, for a time, be part of this plugin.

**Comment by OcelotWalrus** [comment]
Checkout these:
![image](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/05e27889-f0a0-43a3-a81e-dc090f7e9071)
![image](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/19a9682c-2333-42e8-b35f-db8f0cd81c17)

**Reply by lumbar527** [reply]
Is that about how many people have cloned this repo?

(I've done a bunch of cloning because I don't know how to fix conflicts)

**Reply by OcelotWalrus** [reply]
Yes.

Clones or views doesn't matter, it's unique visitors and unique cloners that matter.

**Reply by OcelotWalrus** [reply]
> (I've done a bunch of cloning because I don't know how to fix conflicts)

You don't know how to fix conflicts?
You just have to merge.

**Reply by lumbar527** [reply]
Git stuff I know:
```
git status
git add .
git commit -am "commit name"
git push

git pull

git checkout -b <branch name to make>
git checkout <branch name>

cd path/to/stuff

git clone <repo cloning stuff>
```

**Reply by OcelotWalrus** [reply]
What are the conflicts?

**Reply by lumbar527** [reply]
Usually it's with me committing so there's a different history for local and main. Though when I was trying to add Lumbarian stuff to the cheat stuff, you kept editing it so there were conflicts in that file.

**Reply by OcelotWalrus** [reply]
You should have an error message where it list the possibilities and how to fix it. You should always merge.

**Reply by lumbar527** [reply]
How do you do that? It tells me to do `git pull`, but then I don't understand to do the merge commit.

**Reply by lumbar527** [reply]
OK, sort of figured it out.

**Reply by OcelotWalrus** [reply]
;)

**Comment by OcelotWalrus** [comment]
> Just because this will, for a time, be part of this plugin.

Oh so the species you want to add is the species you told me about yesterday. The `G[][]'[]a[]` species.

**Reply by lumbar527** [reply]
Yep, though I decided to tell you more about them. I'm working on typing that now.

**Comment by lumbar527** [comment]
You know what, I'll just stop the aliens from being secret.

They are the **Guz'Pas**, pronounced "gooz pahz", sort of.

### Home

They are from the Milky Way, but they found a new home further away. (Might remove the further away home part)
Have to figure out a place in the Milky Way, though.

### Technology

They use heavy artillery weaponry with some ammo. (A big turret for a smaller ship might hold 7 rounds)

They mostly use gunboat-size ships.

They're probably tier 2 because their ships are weak health-wize, more like a little stronger than human, but their weaponry is very powerful. They also swarm and have large fleets.

They also have interesting shields. Their shields have a several-second recharge delay, but that does not make as much of a difference. They actually have fairly strong shields compared to their light hull, but they always let 25% of the damage through. They do not need as much energy, though. Now, their shields use is in its resistance to effects: ion, heat, disruption, etc.

### Appearance

They look like bipedal tortoises that are a little taller than humans, but their arms are more like human arms. They have skinnier arms with hands, but they cannot bend them as much as humans can, because of a partial shell covering the elbow and along that side of the arm. These arms are also very heavy, and they can strike you dead in a single blow.

**Reply by lumbar527** [reply]
### History

The Gus'Pas are a relatively new species, only about 6,000 years old (at game start). They developed spaceflight fairly quickly when they discovered their star was going to supernova (B.C. 1,000; 2,000 years old). Only a hundred years later, they met another race, and there was a great war. It lasted 300 years, leaving at a draw when neither could fight more. (the other species may have disappeared by now, though) Even though the Gus'Pas had not made weaponry before the war, they still managed to last that long using big missiles they could scrap together.

By this time, they had learned how to make powerful missiles cheaply. This became their primary defense, and slowly they moved onto the offense. By A.D. 1, they were already spreading out and had made contact and wars with several other aliens. (Maybe they fought the Sheragi?) Then the Great Attack happened, leaving them shrinking.

The Great Attack happened around A.D. 100, when they believed they were attacked by many strange enemies. Several of their major stars went supernova, and ships mysteriously disappeared. (Was it an attack, or was it some anomaly or something that caused this?) After a few hundred more years, they were back to a few systems.

Then they found the wormhole, and went through it. There they found systems and planets to colonize, a secret territory. But while they were colonizing, they found a weaker race that had only a few systems and no hyper/jump drives. There was a war with them that lasted 17 years, ending in the destruction of the weaker race.

After that, they lived in peace but did not expand much, until humans began to expand. Then they destroyed some convoys, but then abandoned some of their systems on the edge. There was another wormhole that connected their space to what would become human space, though the gap between their systems was small enough to use Jump Drives on.

Around A.D. 2,600, the wormhole disappeared. They were now isolated from all other races.

In the time between them leaving and the player starting, the design their shields and begin to build larger transport/cargo ships, but still new warships. (the transport/cargo ships are more medium warship to a little bigger sized.

**Reply by OcelotWalrus** [reply]
I'll read that later :)

**Comment by OcelotWalrus** [comment]
By the way I also edited the map for the Lumbarian colonies system in the Gegno Space.

**Reply by lumbar527** [reply]
I saw, thanks!

**Comment by OcelotWalrus** [comment]
So, just to be sure, for now the new species will be a part of the plugin but you hope to add it to the vanilla?

**Reply by lumbar527** [reply]
Yes.

**Reply by OcelotWalrus** [reply]
okay

**Comment by OcelotWalrus** [comment]
> They are from the Milky Way, but they found a new home further away. (Might remove the further away home part)
Have to figure out a place in the Milky Way, though.

_Currently reading your text_
The space region I could think of for the Gus'Pas are in red.

And the thing in yellow is what I will add later as a 'Quarg Space' just so you know.

![Screenshot from 2023-07-28 16-00-53](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/91f5d49a-ae1a-426c-9aab-0471a342ded6)

**Reply by lumbar527** [reply]
Did you edit the editor? It doesn't have Gegno/Bunrodea space or `republic colors`?

As for the actual locations, thanks!

**Reply by lumbar527** [reply]
Probably north of the Wanderers, then.
References: https://github.com/endless-sky/endless-sky/pull/7518#issuecomment-1304637027, https://github.com/endless-sky/endless-sky/discussions/8806#discussioncomment-6062025

**Comment by OcelotWalrus** [comment]
> Did you edit the editor? It doesn't have Gegno/Bunrodea space or `republic colors`?

Yes. I also asked the author of the plugin author to update the sub-modules, so in the future, the editor will be updated.

I also fixed the bug where it shows no color. In latest endless sky updates, they added a `color` definition that is now used in the governments color definitions, but since the program sources are outdated, it do not recognize the `color` definitions. But I did these modifications only on my AppImage.

**Reply by lumbar527** [reply]
Sometimes I use a self-edited editor with the data files replaced, but that's only editing the `Package Contents`.

**Reply by OcelotWalrus** [reply]
I'm also contributing to the plugin editor to make it better. I coded the `color` definition editor. Do you have any request to make the editor better?

**Reply by lumbar527** [reply]
Yes, do you know how to make it not crash when you save after making a ship?

There are many things that are missing:
adding fleets to systems
adding individual objects to systems

**Comment by OcelotWalrus** [comment]
So how will look the Gus'Pas ships and outfits?

**Reply by lumbar527** [reply]
Ships: like the one posted above; blocky, with big guns.
Outfits: not sure, might work on that soon

**Reply by OcelotWalrus** [reply]
Okay!

**Reply by OcelotWalrus** [reply]
Did you already started coding it or not? Just curious.

**Reply by lumbar527** [reply]
Not yet, though I did make a few systems. I actually am going to put the wormhole systems north of Wanderer space, but the normal systems will be between the Core and Rim, I think

**Comment by OcelotWalrus** [comment]
> Yes, do you know how to make it not crash when you save after making a ship?

No sorry, I didn't know about that.

> There are many things that are missing:
>adding fleets to systems
>adding individual objects to systems

You can already add fleets to systems and add individual objects to systems.

**Reply by OcelotWalrus** [reply]
Maybe a mission/job editor.

**Reply by lumbar527** [reply]
> You can already add fleets to systems and add individual objects to systems.

How? I don't see any button things ore anything in the editor.

> Maybe a mission/job editor.

That would be great, just noting that there already is one that also doesn't support Mac. Can't find it now, will edit soon.

Edit: [Here](https://github.com/shitwolfymakes/Endless-Sky-Mission-Builder)

**Comment by lumbar527** [comment]
Another outfit: Kkrok Shield Emitter
![gus'pas shield medium](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/125756761/efb32b35-2ef8-4c9f-bb87-ec802e842e5a)
Scaled to outfit size, mostly.

**Reply by OcelotWalrus** [reply]
Looks nice.

**Reply by lumbar527** [reply]
Thanks! It's a medium size shield generator.

**Comment by lumbar527** [comment]
Right now, I'm working on this in a separate plugin, here it is if you want to try a few outfits out:
[Guz'Pas.zip](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/files/12207300/Guz.Pas.zip)

**Comment by lumbar527** [comment]
The ship image has changed:
![guz'pas chisel](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/125756761/3ba0b106-684f-43b9-8ed9-1f7151777a14)
When I made it, I didn't realize it was partially transparent. This fixes that.

**Comment by OcelotWalrus** [comment]
I made a cool wallpaper/conversation scene: It's a Quarg Hydra destroying a Lumbarian Met Par Tek 53:
![Hydra Lumbarian](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/a91ead9c-bcdc-4e95-83a6-379f8b0f058c)

**Reply by OcelotWalrus** [reply]
I don't know where to use it in the game...

**Reply by lumbar527** [reply]
How about this: the Lumbarian recruitment mission!
So, you jump into a Lumbarian system (this would be after there is no more Republic, so human space is Lumbarian space). It would probably be near some Cromha thing. The system has a Cutlas-Class (Lumbarian Met Par Tek 53) being attacked by an Empire Hydra! The Cutlas-Class messages you to transmit a message to some Lumbarian fleet. If you do, they will show up after Mr. Timer dies but you also have to evade the Hydra. Then, when you land, you are contacted by a recruiter.

**Reply by OcelotWalrus** [reply]
Sure, could be an idea.

**Comment by OcelotWalrus** [comment]
I re-worked on some outfits to make the textures look better like the Twin Particle Turret, the Guz shields...

**Reply by lumbar527** [reply]
Thanks!

**Comment by OcelotWalrus** [comment]
Hey @lumbar527 , I have a suggestion to propose. Instead of adding all blend files to the `source/` directory of the plugin, I could create a special repository for the sources and then add the sources repository as a sub-module in the plugin directory so that people can choose either to get the sources or not while cloning.

**Reply by lumbar527** [reply]
Sure! Though I didn't know that you could pick if you wanted to clone sub-modules or not.
This plugin is pretty big, as is the `source/` directory.

**Reply by OcelotWalrus** [reply]
> Though I didn't know that you could pick if you wanted to clone sub-modules or not.

By default if you just run `git clone` it won't clone sub-modules, you'll need to run `git clone --recursive` to clone sub-modules as well. Basically sub-modules are just git repositories inside of other git repositories.

**Reply by OcelotWalrus** [reply]
I'm currently doing the sub-module stuff and it might take a while so the source/ directory is not available for now.

**Comment by OcelotWalrus** [comment]
I've found lot of interesting textures. Check them out here:
https://github.com/OcelotWalrus/Cromha-Expansion-sources/tree/c9f68021e646e6c0daa467e17fc43f39151f3bf7/potential%20textures

**Reply by lumbar527** [reply]
The Genesis Flax Cannon could be useful for the Guz, and there are plenty of Quarg things for our Quarg Expansion variant!

**Reply by OcelotWalrus** [reply]
I know!

The Genesis Flax Cannon could be a kind of a fabricator.

**Reply by lumbar527** [reply]
I was actually thinking that the Flax Cannon looked like artillery.

**Reply by OcelotWalrus** [reply]
Actually if you prefer it to be artillery, I'll make it artillery.

**Reply by OcelotWalrus** [reply]
You can start coding it if you want.

**Comment by OcelotWalrus** [comment]
I don't know about you but, for me it sounds insane that 137 different people downloaded the plugin at least once! Some days ago it was just 43.

![image](https://github.com/OcelotWalrus/Cromha-Expansion-plugin/assets/87318892/cc1b8dc6-9b4f-40aa-a2c8-ac95213c73c6)

**Reply by OcelotWalrus** [reply]
I think it's because I updated the description.

It's now
```
Add new high-tier and low-tier factions and expand the existing one. Adds different new storyline and introduce new gameplay mechanics.
```
instead of 
```
Galaxy expansion plugin with various original stuff like new storylines, ships & outfits, new concepts and systems as well as a new galaxy.
```

**Reply by lumbar527** [reply]
Maybe, but it's awesome that that many people are using the plugin!

**Reply by OcelotWalrus** [reply]
Yes.

**Comment by lumbar527** [comment]
What is the Cromha-Plugins organization for? Should I upload some of my plugins to it?

Side note: I think you should change the description from **@ Cromha** to **@ OcelotWalrus**

**Reply by OcelotWalrus** [reply]
> What is the Cromha-Plugins organization for?
For the existing plugins/repositories and for future plugin/projects that have something to do with Endless Sky.

> Side note: I think you should change the description from @ Cromha to @ OcelotWalrus
Yeah you're right.

**Reply by lumbar527** [reply]
> For the existing plugins/repositories and for future plugin/projects that have something to do with Endless Sky.

Or plugins/repositories or yours? Should I put some of my plugins in the organization (Fire Corporation, ES-UI-Upgrades, etc.)?

**Reply by OcelotWalrus** [reply]
Sure.

**Comment by lumbar527** [comment]
I'm creating a project for managing Lumbarian Empire development. Is that OK?

**Reply by OcelotWalrus** [reply]
Sure.

**Comment by lumbar527** [comment]
@OcelotWalrus, do you think you could make my projects public? (For the new repositories)

**Reply by OcelotWalrus** [reply]
Yes.

**Reply by OcelotWalrus** [reply]
I actually can't because there are forks. You need to import to repositories to the organization. I just added you the owner role.

**Reply by lumbar527** [reply]
I think I make the projects public. The repos I want public; they already are.

**Comment by MonkeyMatrix** [comment]
I am starting to work on some spelling and grammatical errors I have found throughout the plugin. When I open pull requests, should I include the commits for one government in each pull request or the entire plugin (where I would just open one pull request)? In addition, I have some questions about various parts of the plugin in regard to this project, so should I create a discussion for this (Spelling/Grammar/Dialog Improvement)? I am new to using Git and GitHub, so I hope I can figure out everything needed for this project without too many hiccups along the way. I have enjoyed exploring this plugin, so thank you for the work put into it. :)

**Reply by lumbar527** [reply]
> When I open pull requests, should I include the commits for one government in each pull request or the entire plugin (where I would just open one pull request)?

I think it's fine to just do one pull request or a bunch, whichever's simpler for you. Unless there is a ton of stuff for one government, I think.

> I have some questions about various parts of the plugin in regard to this project, so should I create a discussion for this (Spelling/Grammar/Dialog Improvement)?

That's fine, I think.

> I have enjoyed exploring this plugin, so thank you for the work put into it. :)

Yeah, Cromha (@OcelotWalrus) is awesome. (He made most of the plugin; I only added a few incomplete things)

Note: As stated, this is @OcelotWalrus's, not mine, so he might have a different request for how you do this.

**Reply by lumbar527** [reply]
Nice to see someone else taking an interest in this!

**Reply by MonkeyMatrix** [reply]
I found it on the endless sky plugin page (I look here frequently for new plugins), and it looked like a pretty large plugin. It turns out it is a massive plugin, and I thought I should help in some way. I found some spelling and grammatical mistakes, so I thought I would start there. I also made a discussion board for this project. Thank you for the work each of you have put into this plugin, and tell me if there is anything I should change on the discussion board posts going forward. :)

**Reply by OcelotWalrus** [reply]
This is fine for me too @lumbar527 .

**Reply by MonkeyMatrix** [reply]
I have added some questions into my discussion about spelling, grammar, dialog, etc. I would appreciate any information or answers you can give to the questions in my Empire Questions comment along with any thoughts on the notable changes. Thank you in advance!

**Comment by MonkeyMatrix** [comment]
I was looking in the empire prototypes ships.txt file, and I was some outfits and effects in it. I am guessing they are related to the black manta right now, but should the outfits be moved the empire outfits file later?

**Reply by OcelotWalrus** [reply]
I don't think so. These outfits are separated because they are not supposed to be obtainable.

**Reply by MonkeyMatrix** [reply]
Thanks for the clarification.

**Comment by OcelotWalrus** [comment]
Hey @lumbar527 ! I've just seen your couples commits on the repository! Thanks for showing interest for the plugin.

I think you should really start working on the Lumbarian Empire storyline/missions! It would add some much more cool content to the plugin.

**Reply by lumbar527** [reply]
Sure! I will begin by rewriting the outline for the discussion!

**Reply by OcelotWalrus** [reply]
Cool!

**Comment by OcelotWalrus** [comment]
I've got a balancing question:
The Empire shielding systems are really OP. But like a lot.
I only talk about the Quatalant Shields and not the Shield Processors.

I'm wondering if I should add something like `shield delay` attribute to the Quatalant Shields.

**Reply by OcelotWalrus** [reply]
There is actually more interesting attributes:
![image](https://github.com/Cromha-Plugins/Cromha-Expansion-plugin/assets/87318892/06a788e7-d8c0-4323-8e45-c0ab5d755bf2)

**Reply by OcelotWalrus** [reply]
Maybe it will be interesting to add some of these attributes to the Shield Processors:
![image](https://github.com/Cromha-Plugins/Cromha-Expansion-plugin/assets/87318892/396ca883-9963-4b3d-b0d2-a09d312726a0)

**Reply by lumbar527** [reply]
Yeah, I think that would be interesting!

**Comment by MonkeyMatrix** [comment]
Recently, you changed the README to recommend people play on the latest unstable endless sky version. I play the game on mobile through endless-mobile. Is there anything I should be aware of when playing on the latest stable version?

**Comment by OcelotWalrus** [comment]
> Recently, you changed the README to recommend people play on the latest unstable endless sky version. I play the game on mobile through endless-mobile. Is there anything I should be aware of when playing on the latest stable version?

It's just that I test things on the continuous build of endless sky because that's the method I use to play the game. I just say that the people should be using the continuous build(of endless sky) while playing the plugin so they are sure there are no problems coming from the endless sky version.

If you're using the latest stable version (v0.0.14), it should be just fine.

**Reply by MonkeyMatrix** [reply]
Makes sense to me. Thanks for the clarification!

**Comment by MonkeyMatrix** [comment]
I saw this [page](https://github.com/Cromha-Plugins/Cromha-Expansion-plugin/wiki/User-Guide) on the wiki, and I think it would be helpful for instructions to be given to those playing [endless-mobile](https://github.com/thewierdnut/endless-mobile) (which I also play). I was unsure of how to contribute to the wiki, so I thought I would post here. I can move this to an issue if you think it would be useful to players. Something like:

---

If you are playing Endless Sky on the [mobile](https://github.com/thewierdnut/endless-mobile) version for Android, you can follow one of two steps to download the plugin. You can either use `git clone` in [Termux](https://termux.dev/en/) (or another terminal emulator for Android) or download the plugin from the [releases](https://github.com/Cromha-Plugins/Cromha-Expansion-plugin/releases) page.



Using a terminal emulator such as [Termux](https://termux.dev/en/):
- `git clone https://github.com/OcelotWalrus/Cromha-Expansion-plugin.git` in the terminal emulator.
- Open Endless Sky.
- Navigate to Preferences > Plugins.
- Press "Import."
- Navigate to the compressed file (usually located in your downloads folder).
- Open the file.
- Follow the prompts.
- Enjoy the plugin! (Or any other plugins!)



Using the [releases](https://github.com/Cromha-Plugins/Cromha-Expansion-plugin/releases) page:
- Navigate to the [releases](https://github.com/Cromha-Plugins/Cromha-Expansion-plugin/releases) page.
- Choose a continuous version (may be unstable) or a stable version (which has a version number and is marked with "Stable").
- Download either of the .zip or .tar.gz files (You will get the same result.).
- Open Endless Sky.
- Navigate to Preferences > Plugins.
- Press "Import."
- Navigate to the compressed file (usually located in your downloads folder).
- Open the file.
- Follow the prompts.
- Enjoy the plugin! (Or any other plugins!)

Updating the plugin is as simple as navigating to the cloned directory in your terminal emulator and using `git pull` or downloading a newer release on the [releases](https://github.com/Cromha-Plugins/Cromha-Expansion-plugin/releases) page, and then you should follow the above steps.

Note: If the game crashes on startup, this could indicate your device is not powerful enough to run with the plugin installed. If you have any other plugins enabled, you could try disabling or removing (preferably disabling) them (**This can break your saves!**).

---

I was unsure of the duplicated bullet list in the two downloading sections. I think separating the different downloading methods would work as well.

**Reply by lumbar527** [reply]
I've added it to the wiki.

**Reply by MonkeyMatrix** [reply]
Looks great!

**Comment by MonkeyMatrix** [comment]
I would appreciate someone looking at a few commits in my fork to make sure I am doing this correctly.
https://github.com/MonkeyMatrix/Cromha-Expansion-plugin

**Reply by MonkeyMatrix** [reply]
In this fork, I had apparently not changed colonies to Colonies for the Mineria Colonies yet. Once I fix some of the values which got changed for outfit stats (I don't know why they got changed), I will push that change.

**Comment by MonkeyMatrix** [comment]
I was wondering if there was a reason we use "payment" and then "payment <amount>" instead of just "payment <amount>." I do not know much about developing plugins, and I thought it seemed interesting.

https://github.com/Cromha-Plugins/Cromha-Expansion-plugin/blob/f4618b48c1ce66836054ac51abc63930e9e11272/data/Empire/empire%20storyline%200.txt#L95-L96

**Reply by OcelotWalrus** [reply]
I don't think that's necessary, just a bad habit. But it's not necessary to remove it if you want.
